pkg/cmd/grafana/apiserver/auth/tokenAuthenticator.go

```diff
@@ -0,0 +1,71 @@
+package auth
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/grafana/authlib/authn"
+	"k8s.io/apiserver/pkg/authentication/authenticator"
+	"k8s.io/apiserver/pkg/authentication/user"
+)
+
+const (
+	headerKeyAccessToken = "X-Access-Token"
+	headerKeyGrafanaID   = "X-Grafana-Id"
+
+	extraKeyAccessToken = "access-token"
+	extraKeyGrafanaID   = "id-token"
```

Comment:
I was gonna say this yesterday but good you decided to keep the extra key name as `id-token`. Since we also have this setup here as `id-token` in the embedded apiserver's authenticator: https://github.com/grafana/grafana/blob/main/pkg/services/apiserver/auth/authenticator/signedinuser.go#L39

Reply:
I think we may actually want https://github.com/grafana/grafana/blob/v10.4.1/pkg/services/auth/identity/requester.go#L22 or signedInUser should directly implement user.Info